test/e2e/framework.go

```diff
@@ -419,3 +420,136 @@ func kubectlExec(namespace string, podName, containerName string, args ...string
 func KubeDescribe(text string, body func()) bool {
 	return Describe("[k8s.io] "+text, body)
 }
+
+// PodStateVerification represents a verification of pod state.
+// Any time you have a set of pods that you want to operate against or query,
+// this struct can be used to declaratively identify those pods.
+type PodStateVerification struct {
+	// Optional: only pods that have k=v labels will pass this filter.
+	selectors map[string]string
```

Comment:
the reason i didnt break it down further, was because this way you can do `PodStateVerification{map[string]string{selectorKey: selectorValue}..., []api.PodPhase{api.PodRunning}` super concise state definitions for the most common use cases.  maybe modularize further in a future iteration?